tests/unit/test_lib/test_utilities/test_filters/test_windowing_scheme.py

```diff
@@ -0,0 +1,90 @@
+#  Redistribution and use in source and binary forms, with or without         #
+#  modification, are permitted provided that the following conditions are met:#
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import time
+import unittest
+
+from liota.lib.utilities.filters.range_filter import RangeFilter, Type
+from liota.lib.utilities.filters.windowing_scheme.windowing_scheme import WindowingScheme
+
+class WindowingSchemeTest(unittest.TestCase):
+    """
+    Unit test cases for Windowing Scheme
+    """
+    def setUp(self):
+        """Initialize lower bound, upper bound and a filter for Windowing scheme test"""
+        self.lower_bound = 10
+        self.upper_bound = 20
+        self.window_test_filter = RangeFilter(Type.CLOSED, self.lower_bound, self.upper_bound)
+
+
+    def test_init_windowing_scheme_filter(self):
+        """If filter not provided in Windowing scheme, raise TypeError"""
+        self.assertRaises(TypeError, lambda: WindowingScheme("", 10))
+
+    def test_init_windowing_scheme_window_size(self):
+        """If window size is not a non negative number, raise ValueError"""
+        self.assertRaises(ValueError, lambda: WindowingScheme(self.window_test_filter, "10"))
+        self.assertRaises(ValueError, lambda: WindowingScheme(self.window_test_filter, -10))
+
+    def test_window_filter(self):
+        """If next window time has not elapsed, pass the sample"""
```

Comment:
Change comment to "If filter returns a valid value, windowing_scheme should return it"